assessment/collectors/app_migration_analysis.go

```diff
@@ -123,125 +128,124 @@ func NewMigrationSummarizer(ctx context.Context, googleGenerativeAIAPIKey *strin
-		sourceFramework:               sourceFramework,
-		targetFramework:               targetFramework,
-		targetSchema:                  targetSchema,
-		projectPath:                   projectPath,
-		dependencyGraph:               make(map[string]map[string]struct{}),
-		fileDependencyAnalysisDataMap: make(map[string]FileDependencyAnalysisData),
-	}
-	m.modelFlash.ResponseMIMEType = "application/json"
-	m.modelPro.ResponseMIMEType = "application/json"
-
-	return m, nil
+	summarizer := &MigrationCodeSummarizer{
+		gcpProjectID:              projectID,
+		gcpLocation:               location,
+		aiClient:                  client,
+		geminiProModel:            client.GenerativeModel("gemini-1.5-pro-002"),
+		geminiFlashModel:          client.GenerativeModel("gemini-2.0-flash-001"),
+		conceptExampleDatabase:    conceptExampleDB,
+		projectDependencyAnalyzer: dependencyAnalyzer.AnalyzerFactory(language, ctx),
+		sourceDatabaseSchema:      sourceSchema,
+		sourceDatabaseFramework:   sourceFramework,
+		targetDatabaseFramework:   targetFramework,
+		targetDatabaseSchema:      targetSchema,
+		projectRootPath:           projectPath,
+		dependencyGraph:           make(map[string]map[string]struct{}),
+		fileDependencyAnalysis:    make(map[string]FileDependencyInfo),
+	}
+	summarizer.geminiFlashModel.ResponseMIMEType = "application/json"
+	summarizer.geminiProModel.ResponseMIMEType = "application/json"
+
+	return summarizer, nil
 }
 
-func (m *MigrationSummarizer) MigrationCodeConversionInvoke(
+// InvokeCodeConversion performs code conversion using the LLM.
+func (m *MigrationCodeSummarizer) InvokeCodeConversion(
 	ctx context.Context,
 	originalPrompt, sourceCode, olderSchema, newSchema, identifier string,
 ) (string, error) {
-	prompt := AnalyzeCodePromptTemplate
-	prompt = strings.ReplaceAll(prompt, "{{SOURCE_FRAMEWORK}}", m.sourceFramework)
-	prompt = strings.ReplaceAll(prompt, "{{TARGET_FRAMEWORK}}", m.targetFramework)
+	prompt := analyzeCodePromptTemplate
+	prompt = strings.ReplaceAll(prompt, "{{SOURCE_FRAMEWORK}}", m.sourceDatabaseFramework)
+	prompt = strings.ReplaceAll(prompt, "{{TARGET_FRAMEWORK}}", m.targetDatabaseFramework)
 	prompt = strings.ReplaceAll(prompt, "{{SOURCE_CODE}}", sourceCode)
 	prompt = strings.ReplaceAll(prompt, "{{OLDER_SCHEMA}}", olderSchema)
 	prompt = strings.ReplaceAll(prompt, "{{NEW_SCHEMA}}", newSchema)
 
-	resp, err := m.modelFlash.GenerateContent(ctx, genai.Text(prompt))
+	response, err := m.geminiFlashModel.GenerateContent(ctx, genai.Text(prompt))
 	if err != nil {
 		return "", err
 	}
-	logger.Log.Debug("Token: ",
-		zap.Int32("Prompt Token:", resp.UsageMetadata.PromptTokenCount),
-		zap.Int32("Candidate Token:", resp.UsageMetadata.CandidatesTokenCount),
-		zap.Int32("Total Token:", resp.UsageMetadata.TotalTokenCount))
-
-	var response string
-	if p, ok := resp.Candidates[0].Content.Parts[0].(genai.Text); ok {
-		response = string(p)
+	logger.Log.Debug("LLM Token Usage (Initial Conversion): ",
```

Comment:
make this an info log. This is something the users would be interested in.

Alternately, It would be good if we can share this as an output of the assessment overall.